src/main/java/org/zendesk/client/v2/Zendesk.java

```diff
@@ -1344,6 +1345,84 @@ public void changeUserPassword(User user, String oldPassword, String newPassword
             handleStatus()));
   }
 
+  public List<UserProfile> getUserProfilesForUser(User user) {
+    return getUserProfilesForUser(user.getId());
+  }
+
+  public List<UserProfile> getUserProfilesForUser(long userId) {
+    return complete(
+        submit(
+            req("GET", tmpl("/users/{user_id}/profiles").set("user_id", userId)),
+            handleList(UserProfile.class, "profiles")));
+  }
+
+  public UserProfile getUserProfile(UserProfile userProfile) {
+    return getUserProfile(userProfile.getId());
+  }
+
+  public UserProfile getUserProfile(String userProfileId) {
+    return complete(
+        submit(
+            req("GET", tmpl("/user_profiles/{profile_id}").set("profile_id", userProfileId)),
+            handle(UserProfile.class, "profile")));
+  }
+
+  public UserProfile getUserProfilebyIdentifier(String identifier) {
+    return complete(
+        submit(
+            req(
+                "GET",
+                tmpl("/user_profiles?identifier={identifier}").set("identifier", identifier)),
+            handle(UserProfile.class, "profile")));
+  }
+
+  public UserProfile createOrUpdateUserProfile(UserProfile userProfile) {
+    return createOrUpdateUserProfile(
+        userProfile,
+        userProfile.getIdentifiers().get(0).getType(),
+        userProfile.getIdentifiers().get(0).getValue());
+  }
+
+  public UserProfile createOrUpdateUserProfile(
+      UserProfile userProfile, String identifierType, String identifierValue) {
```

Comment:
what about aligning with the `UserProfile` object and use an `Identifier` object here?